source/common/grpc/json_transcoder_filter.cc

```diff
@@ -0,0 +1,397 @@
+      throw EnvoyException("transcoding_filter: Unable to parse proto descriptor");
+    }
+  }
+
+  // TODO(lizan): Consider factor out building PathMatcher building.
+  google::grpc::transcoding::PathMatcherBuilder<const google::protobuf::MethodDescriptor*> pmb;
+
+  for (const auto& service_name : config.getStringArray("services")) {
+    auto service = descriptor_pool_.FindServiceByName(service_name);
+    if (service == nullptr) {
+      throw EnvoyException("transcoding_filter: Could not find '" + service_name +
+                           "' in the proto descriptor");
+    }
+    for (int i = 0; i < service->method_count(); ++i) {
+      auto method = service->method(i);
+
+      auto http_rule = method->options().GetExtension(google::api::http);
+
+      switch (http_rule.pattern_case()) {
+      case ::google::api::HttpRule::kGet:
+        pmb.Register("GET", http_rule.get(), http_rule.body(), method);
+        break;
+      case ::google::api::HttpRule::kPut:
+        pmb.Register("PUT", http_rule.put(), http_rule.body(), method);
+        break;
+      case ::google::api::HttpRule::kPost:
+        pmb.Register("POST", http_rule.post(), http_rule.body(), method);
+        break;
+      case ::google::api::HttpRule::kDelete:
+        pmb.Register("DELETE", http_rule.delete_(), http_rule.body(), method);
+        break;
+      case ::google::api::HttpRule::kPatch:
+        pmb.Register("PATCH", http_rule.patch(), http_rule.body(), method);
+        break;
+      case ::google::api::HttpRule::kCustom:
+        pmb.Register(http_rule.custom().kind(), http_rule.custom().path(), http_rule.body(),
+                     method);
+        break;
+      default:
+        break;
+      }
+    }
+  }
+
+  path_matcher_ = pmb.Build();
+
+  type_helper_.reset(new google::grpc::transcoding::TypeHelper(
+      google::protobuf::util::NewTypeResolverForDescriptorPool(TypeUrlPrefix, &descriptor_pool_)));
+}
+
+Status JsonTranscoderConfig::createTranscoder(
+    const Http::HeaderMap& headers, ZeroCopyInputStream& request_input,
+    google::grpc::transcoding::TranscoderInputStream& response_input,
+    std::unique_ptr<Transcoder>& transcoder,
+    const google::protobuf::MethodDescriptor*& method_descriptor) {
+  const std::string method = headers.Method()->value().c_str();
+  std::string path = headers.Path()->value().c_str();
+  std::string args;
+
+  const size_t pos = path.find('?');
+  if (pos != std::string::npos) {
+    args = path.substr(pos + 1);
+    path = path.substr(0, pos);
+  }
+
+  RequestInfo request_info;
+  std::vector<VariableBinding> variable_bindings;
+  method_descriptor =
+      path_matcher_->Lookup(method, path, args, &variable_bindings, &request_info.body_field_path);
+  if (!method_descriptor) {
+    return Status(Code::NOT_FOUND, "Could not resolve " + path + " to a method");
+  }
+
+  Status status = methodToRequestInfo(method_descriptor, &request_info);
+  if (!status.ok()) {
+    return status;
```

Comment:
100% coverage is expected when reasonable. In this case this PR is big enough already. Let's do a follow up but please open an issue and link it here so that we can track.